### PR TITLE
Enable Document Type Declaration Serialization

### PR DESCRIPTION
**Pull request by adamretter**
This PR now correctly enables the serialisation of Document Type Declarations if one is present in a document. Previously this behaviour was inconsistent across eXist-db, some APIs would serialise the DocType which others would not.

This has been implemented and tested for the following APIs:
* HTTP RESTServer
* XML-RPC
* XML:DB Local and Remote
* WebDAV

As part of the document, the Document Type Declaration is now visible and modifiable from:
* eXist-db Java Admin Client
* Oxygen XML Editor
* eXide

Closes https://github.com/eXist-db/exist/issues/2056
Related to https://github.com/eXist-db/eXide/issues/198

### CODE REVIEW

**Comment by joewiz** [comment]
Is this option now available in serialization option declarations in query prologs? If so, should an entry be added under https://exist-db.org/exist/apps/doc/xquery#legacy-serialization?

As to the question of classifying the changes in this PR, recall that in a related thread on exist-open, https://markmail.org/message/w4g4kt3o4gh5dvy6, @wolfgangmm noted that eXist was already storing doctype information, but wasn't serializing it. This PR's addition of doctype serialization and enabling it by default in all APIs strikes me as something between a bugfix and a comprehensively implemented new feature—providing something that users have often expected, requested, and struggled to overcome. I have struggled to imagine any process that would *break* if a doctype were supplied rather than omitted. This would argue in favor of including it in 6.x, rather than requiring a new major version.

**Review by dizzzz — APPROVED** [review]

**Comment by adamretter** [comment]
> Is this option now available in serialization option declarations in query prologs? If so, should an entry be added under https://exist-db.org/exist/apps/doc/xquery#legacy-serialization?

Good idea - I have opened a PR for that here - https://github.com/eXist-db/documentation/pull/829

> that users have often expected, requested, and struggled to overcome

That was indeed my own experience recently!

It seems to me that people confuse XML documents and their serialization, with XDM Serialization. As I understand it XML documents can have Doctype Declarations with no issue, whereas XDM has no concept of a Doctype Declaration, and Serialization of an XDM (i.e the result of an XQuery) only allows you to statically declare a doctype as a serialization option (unless using `fn:serialize` within an XDM).

This PR focuses on XML serialization through eXist-db's various APIs.

Additionally it does add an (eXist-db specific) serialization option for XDM via `fn:serialize` to output the DocType if you happen to be serializing a persistent document from the database, otherwise that option has no effect.

> This would argue in favor of including it in 6.x, rather than requiring a new major version.

Sounds good to me :-)

File: exist-core/src/main/java/org/exist/client/InteractiveClient.java [hunk 1]
```diff
@@ -137,6 +138,7 @@ public class InteractiveClient {
         defaultProps.setProperty(USER, USER_DEFAULT);
         defaultProps.setProperty(EDITOR, EDIT_CMD);
         defaultProps.setProperty(INDENT, "true");
+        defaultProps.setProperty(OUTPUT_DOCTYPE, "true");
```
Comment (line-o) [inline exist-core/src/main/java/org/exist/client/InteractiveClient.java:141]: Could this break existing applications or was the default to output doctype declarations true by default before?

**Review by line-o — APPROVED** [review]
The changes do look very good and I am looking forward to when this is pulled in.
Having written that, I am curious if this might be a breaking change or not.

**Pull request merged**